### PR TITLE
Give a good error message on unassigned ports 

### LINKED ISSUE #22552: NPE in GraphInterpreter when `GraphStageLogic` never called `setHandler`
See https://gitter.im/akka/dev?at=58c7d6ebfef1d2de52ad22b4

It seems the nicer error message we had with the old materializer was lost.

Example:

```scala
      Source.empty[Unit]
        .fold("") { case _ => "" + "" }
        .viaMat(new GraphStageWithMaterializedValue[FlowShape[String, String], String] {
          val in = Inlet[String]("in")
          val out = Outlet[String]("out")
          override val shape: FlowShape[String, String] = FlowShape(in, out)

          override def createLogicAndMaterializedValue(inheritedAttributes: Attributes) =
            new GraphStageLogic(shape) with InHandler with OutHandler {
              override def onPush(): Unit = push(out, grab(in))

              override def onPull(): Unit = pull(in)
            } -> "MAT"
        })(Keep.right)
        .to(Sink.head)
```

```
[info] java.lang.NullPointerException
[info]     at akka.stream.impl.fusing.GraphInterpreter.processPull(GraphInterpreter.scala:507)
[info]     at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:467)
[info]     at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:369)
[info]     at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:570)
[info]     at akka.stream.impl.fusing.GraphInterpreterShell.init(ActorGraphInterpreter.scala:540)
[info]     at akka.stream.impl.fusing.ActorGraphInterpreter.tryInit(ActorGraphInterpreter.scala:658)
[info]     at akka.stream.impl.fusing.ActorGraphInterpreter.preStart(ActorGraphInterpreter.scala:706)
[info]     at akka.actor.Actor$class.aroundPreStart(Actor.scala:521)
[info]     at akka.stream.impl.fusing.ActorGraphInterpreter.aroundPreStart(ActorGraphInterpreter.scala:649)
[info]     at akka.actor.ActorCell.create(ActorCell.scala:574)
[info]     at akka.actor.ActorCell.invokeAll$1(ActorCell.scala:445)
[info]     at akka.actor.ActorCell.systemInvoke(ActorCell.scala:467)
[info]     at akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:282)
[info]     at akka.dispatch.Mailbox.run(Mailbox.scala:223)
[info]     at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
[info]     at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
[info]     at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
[info]     at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
[info]     at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #22552

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala [hunk 1]
```diff
@@ -600,6 +600,17 @@ final class GraphStageIsland(
...
+    while (idx < logic.handlers.length) {
+      if (logic.handlers(idx).asInstanceOf[AnyRef] eq null) {
+        val port =
+          if (idx < logic.inCount) stage.shape.inlets(idx)
+          else stage.shape.outlets(idx - logic.inCount)
+        throw new IllegalStateException(s"No handler defined in stage [$stage] for port [$port]." +
+          " All inlets and outlets must be assigned a handler with setHandler in the constructor of your graph stage logic.")
+      }
+      idx += 1
+    }
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala]: an alternative might be to do it in a catch of NPE?

Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala]: That would be in the interpreter then, I kind of feel this should fail already in materialization.

Pushed alternative solution without loop for comparison.

Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala]: ah, I thought the exception was already thrown from here

**Review by ktoso — APPROVED** [review]
I like the latest revision (with the checks inside the materializer). LGTM, curious on perf number

**Comment by patriknw** [comment]
I have verified bench results after rebasing on https://github.com/akka/akka/pull/22480 and it's looking good. 140k/s

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**